canonicalwebteam/search/views.py

```diff
@@ -0,0 +1,69 @@
+# Standard library
+import os
+
+# Packages
+import flask
+
+# Local
+from canonicalwebteam.search.models import get_search_results
+
+
+class NoAPIKeyError(Exception):
+    pass
+
+
+def build_search_view(site=None, template_path="search.html"):
+    """
+    Build and return a view function that will query the
+    Google Custom Search API and then render search results
+    using the provided template.
+
+    Usage in e.g. `app.py`:
+
+        from canonicalwebteam.search import build_search_view
+
+        app = Flask()
+        app.add_url_rule(
+            "/search",
+            "search",
+            build_search_view(
+                site="snapcraft.io",
+                template_path="search.html"
+            )
+        )
+    """
+
+    def search_view():
+        """
+        Get search results from Google Custom Search
+        """
+
+        # The webteam's default custom search ID
+        search_engine_id = "009048213575199080868:i3zoqdwqk8o"
```

Comment:
should we put this in the projects configurations instead of here? In case we have to update it we won't have to update the extension itself.

Reply:
I did put it here deliberately, to minimise what you need to configure on the project. Since we do use the same key everywhere, and if we change it we will want to update it centrally.

I could make it so this was simply the default and it was over-rideable at the project level, but it feels like overengineering when we don't have a single project that uses a different key.